datumaro/components/media.py

```diff
@@ -530,3 +532,36 @@ def __init__(
     @property
     def data(self) -> List[Image]:
         return self._images
+
+
+class RoIImage(Image):
+    def __init__(
+        self,
+        data: Union[np.ndarray, Callable[[str], np.ndarray], None] = None,
+        *,
+        path: Optional[str] = None,
+        ext: Optional[str] = None,
+        roi: BboxIntCoords,
+    ) -> None:
+        self._roi = roi
```

Comment:
don't we need have any data type conversion when getting floating-point roi with the warning message?

Reply:
or error message for this.

Reply:
Added an `assert`.